packages/flutter/lib/src/rendering/sliver_fixed_extent_list.dart

```diff
@@ -234,6 +255,20 @@ abstract class RenderSliverFixedExtentBoxAdaptor extends RenderSliverMultiBoxAda
 
   late SliverLayoutDimensions _currentLayoutDimensions;
 
+  // Used by itemExtentBuilder to prevent out-of-bounds access.
+  int? _childCount;
+  void _cacheChildCountIfNeeded() {
```

Comment:
Oh this is neat, but I don't know if it is necessary. The scenarios I was referring to are when the SliverChildDelegate gives us a fixed childCount. 
Such as SliverChildBuilderDelegate.childCount, and SliverChildListDelegate.children.length.

If the sliver does not know how many children it has, then I would totally expect the extent builder to just return null, just like when the child builder delegate returns null to indicate the end of the list when childCount is not set. 

Do you think we need to try to cache this extent? The implementation of isFiniteChildren could be very expensive. Do you think we need it?